src/main/java/org/jabref/gui/texparser/WalkFileTreeTask.java

```diff
@@ -0,0 +1,73 @@
+package org.jabref.gui.texparser;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.jabref.gui.util.BackgroundTask;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WalkFileTreeTask extends BackgroundTask<FileNodeViewModel> {
```

Comment:
Maybe rename to `SearchTexFilesTask` ? Moreover, since you wrap the call method in a `Backgroundtask` anyway, there is no need to extend here from `BackgroundTask`. You might even want to consider moving the `searchDirectory` method to `ParseTexDialogViewModel ` and remove this class completely.